src/base/Roles/AccountantWithRateProviders.sol

```diff
@@ -283,6 +322,54 @@ contract AccountantWithRateProviders is Auth, IRateProvider {
         emit ExchangeRateUpdated(uint96(currentExchangeRate), newExchangeRate, currentTime);
     }
 
+    /**
+     * @notice Set lending rate (expensive - requires checkpoint)
+     */
+    function setLendingRate(uint256 _lendingRate) external requiresAuth {
+        require(_lendingRate <= maxLendingRate, "Lending rate exceeds maximum");
+        // Checkpoint if deposits exist
+        if (vault.totalSupply() > 0 && lendingInfo.lendingRate > 0) {
+            uint256 timeElapsed = block.timestamp - lendingInfo.lastAccrualTime;
+
+            // Calculate interest
+            uint256 totalDeposits = vault.totalSupply().mulDivDown(accountantState.exchangeRate, ONE_SHARE);
+            uint256 interestAccrued =
+                totalDeposits.mulDivDown(lendingInfo.lendingRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+
+            // Protocol fees
+            uint256 protocolFees =
+                totalDeposits.mulDivDown(lendingInfo.protocolFeeRate * timeElapsed, SECONDS_PER_YEAR * BASIS_POINTS);
+            accountantState.feesOwedInBase += uint128(protocolFees);
+
+            // Update exchange rate
+            if (vault.totalSupply() > 0) {
+                uint256 rateIncrease = interestAccrued.mulDivDown(ONE_SHARE, vault.totalSupply());
+                accountantState.exchangeRate += uint96(rateIncrease);
+            }
+        }
+
+        lendingInfo.lendingRate = _lendingRate;
+        lendingInfo.lastAccrualTime = block.timestamp;
+        emit LendingRateUpdated(_lendingRate, block.timestamp);
+    }
+
+    /**
+     * @notice Set protocol fee rate (cheap - no checkpoint)
+     */
+    function setProtocolFeeRate(uint256 _protocolFeeRate) external requiresAuth {
+        lendingInfo.protocolFeeRate = _protocolFeeRate;
```

Comment:
Unlike updateExchangeRate() and setLendingRate(), I believe setProtocolFeeRate() should accrue protocol fees before setting the new rate to ensure accounting is consistent.